File: src/types/agents.ts
/** Agent role identifiers for the orchestration pipeline. */
export type AgentRole =
  | "prompt_enhancer"
  | "planner"
  | "plan_auditor"
  | "coder"
  | "reviewer_auditor"
  | "code_fixer"
  | "judge"
  | "executive_summary";

/** Supported CLI agent backends. */
export type AgentBackend = "claude" | "codex" | "gemini" | "kimi" | "copilot" | "opencode";

/** CLI health check result returned from the backend. */
export interface CliHealth {
  claude: CliStatus;
  codex: CliStatus;
  gemini: CliStatus;
  kimi: CliStatus;
  copilot: CliStatus;
  opencode: CliStatus;
}

/** Availability status for a single CLI tool. */
export interface CliStatus {
  available: boolean;
  path: string;
  error?: string;
}

/** Version and availability information for a single CLI tool. */
export interface CliVersionInfo {
  name: string;
  cliName: string;
  installedVersion?: string;
  latestVersion?: string;
  upToDate: boolean;
  updateCommand: string;
  available: boolean;
  error?: string;
}

/** Aggregate version information for all CLI tools. */
export interface AllCliVersions {
  claude: CliVersionInfo;
  codex: CliVersionInfo;
  gemini: CliVersionInfo;
  kimi: CliVersionInfo;
  copilot: CliVersionInfo;
  opencode: CliVersionInfo;
}

File: src-tauri/src/models/agents.rs
use serde::{Deserialize, Serialize};

/// Agent role identifiers for the orchestration pipeline.
#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "snake_case")]
pub enum AgentRole {
    PromptEnhancer,
    Planner,
    PlanAuditor,
    Coder,
    ReviewerAuditor,
    CodeFixer,
    Judge,
    ExecutiveSummary,
}

/// Supported CLI agent backends.
#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "lowercase")]
pub enum AgentBackend {
    Claude,
    Codex,
    Gemini,
    Kimi,
    Copilot,
    OpenCode,
}

pub(crate) fn default_prompt_enhancer_agent() -> AgentBackend {
    AgentBackend::Claude
}

pub(crate) fn default_executive_summary_agent() -> AgentBackend {
    AgentBackend::Codex
}

pub(crate) fn default_executive_summary_model() -> String {
    "codex-5.3".to_string()
}

pub(crate) fn default_kimi_path() -> String {
    "kimi".to_string()
}

pub(crate) fn default_opencode_path() -> String {
    "opencode".to_string()
}

pub(crate) fn default_copilot_path() -> String {
    "gh".to_string()
}

pub(crate) fn default_kimi_model() -> String {
    "kimi-k2.5".to_string()
}

pub(crate) fn default_opencode_model() -> String {
    "opencode/glm-5".to_string()
}

pub(crate) fn default_copilot_model() -> String {
    "default".to_string()
}
